### PR TITLE
Flower Pots & More Flowers

### PR DESCRIPTION
**Pull request by cosmiccoincidence**
#### Summary
 ```SUMMARY: Content "Adds flower pots and ways/places to get them in game."```

#### Purpose of change
This change adds flower pots as a precursor so if anyone wants to add planting in pots.
Also adds 4 flowers (rose, hydrangea, tulip, and spurge).

Also linted some files.

Dependent on #25360.

Added plants to regional_map_settings but will add plants to specific locations in a separate PR after (adding these plants and the ones from my other location, at the same time).

### CODE REVIEW

**Comment by kevingranade** [comment]
> Also linted all files that I touched, hence the large number of lines changed.

Please stop doing this, it makes your PRs difficult to merge.

File: data/json/furniture.json [hunk 1]
```diff
@@ -2055,6 +2119,30 @@
...
+  {
+    "type": "furniture",
+    "id": "f_flower_tulip",
+    "name": "tulip",
+    "symbol": "f",
+    "color": "light_magenta",
+    "move_cost_mod": 0,
+    "required_str": -1,
+    "flags": [ "TRANSPARENT", "TINY", "FLAMMABLE_ASH" ],
+    "examine_action": "flower_tulip",
```
Comment (cosmiccoincidence) [inline data/json/furniture.json]: For some reason my examine actions are not working... I'm getting an error.

**Comment by cosmiccoincidence** [comment]
I can do that.

File: data/json/terrain.json [hunk 2]
```diff
@@ -3836,6 +3836,84 @@
...
+    "type": "terrain",
+    "id": "t_shrub_rose",
+    "name": "rose bush",
+    "symbol": "#",
+    "color": "light_red_green",
+    "move_cost": 8,
+    "flags": [ "TRANSPARENT", "CONTAINER", "FLAMMABLE_ASH", "THIN_OBSTACLE", "SHRUB", "SHORT" ],
+    "transforms_into": "t_shrub_rose_harvested",
+    "examine_action": "harvest_ter_nectar",
+    "//": "Insert rose harvest once flowers have a use (same for other generic flowers)."
```
Comment (ZhilkinSerg) [inline data/json/terrain.json]: Missing comma at end of line.

File: data/json/terrain.json [hunk 3]
```diff
@@ -3836,6 +3836,84 @@
...
+    "type": "terrain",
+    "id": "t_shrub_hydrangea",
+    "name": "hydrangea bush",
+    "symbol": "#",
+    "color": "light_blue_green",
+    "move_cost": 8,
+    "flags": [ "TRANSPARENT", "CONTAINER", "FLAMMABLE_ASH", "THIN_OBSTACLE", "SHRUB", "SHORT" ],
+    "transforms_into": "t_shrub_hydrangea_harvested",
+    "examine_action": "harvest_ter_nectar",
+    "//": "Insert hydrangea harvest once flowers have a use (same for other generic flowers)."
```
Comment (ZhilkinSerg) [inline data/json/terrain.json]: Missing comma at end of line.

File: src/iexamine.cpp [hunk 4]
```diff
@@ -1420,6 +1420,38 @@ void iexamine::flower_bluebell(player &p, const tripoint &examp)
     return;
 }
 
+/**
+ * It's a flower, drink nectar if your able to.
+ */
+void iexamine::flower_tulip(player &p, const tripoint &examp)
```
Comment (ZhilkinSerg) [inline src/iexamine.cpp]: You need to put new flower function declarations to `iexamine.h`:

```CPP
void iexamine::flower_tulip(player &p, const tripoint &examp)
void iexamine::flower_spurge(player &p, const tripoint &examp)
```

Comment (cosmiccoincidence) [reply src/iexamine.cpp]: Thanks, I figured i missed somthing somewhere.

**Comment by cosmiccoincidence** [comment]
>Roses, but no rose hips?

I got more berries added in another PR currently open (#25360), Ill add rose hips to that PR.

**Comment by cosmiccoincidence** [comment]
Reverted the linting of item_groups.json as this PR started to have conflicts due to it.

File: data/json/regional_map_settings.json [hunk 5]
```diff
@@ -24,6 +28,10 @@
           "boosted_other": {                "//": "...and the first non-default picked from this list...",
               "t_shrub_blueberry": 40.0,
               "t_shrub_strawberry": 30.0,
+              "t_shrub_rose": 5.0,
+              "t_shrub_hydrangea": 3.5,
+              "f_flower_tulip": 3.5,
+              "f_flower_spurge": 5.0,
```
Comment (ZhilkinSerg) [inline data/json/regional_map_settings.json]: image

File: data/json/terrain.json [hunk 6]
```diff
@@ -3836,6 +3836,89 @@
...
+    "id": "t_shrub_rose",
+    "name": "rose bush",
+    "description": "A fat bush of beautiful red roses, if only you could get a date!  Watch out for its thorns!",
+    "symbol": "#",
+    "color": "light_red_green",
+    "move_cost": 8,
+    "flags": [ "TRANSPARENT", "CONTAINER", "FLAMMABLE_ASH", "THIN_OBSTACLE", "SHRUB", "SHORT", "SHARP" ],
+    "transforms_into": "t_shrub_rose_harvested",
+    "examine_action": "harvest_ter_nectar",
+    "harvest_by_season": [ { "seasons": [ "autumn" ], "entries": [ { "drop": "rose_hips", "base_num": [ 2, 5 ], "scaled_num": [ 0, 0.5 ] } ] } ],
```
Comment (ZhilkinSerg) [inline data/json/terrain.json]: I guess that is from #25360, right?

image

File: data/json/regional_map_settings.json (hunk 5 above)
Comment (cosmiccoincidence) [reply data/json/regional_map_settings.json]: This _should_ be good once the other PR is merged first as I lowered these numbers in the other PR.

File: data/json/terrain.json (hunk 6 above)
Comment (cosmiccoincidence) [reply data/json/terrain.json]: Yeah, it's trying to drop rose hips where are the fruit after the flower and I added those in the other PR.

**Comment by cosmiccoincidence** [comment]
It still says conflicting files... maybe because I just manually added the missing data?

**Pull request merged**